src/UI/Component/Chart/Speedo/Factory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/* Copyright (c) 2017 Ralph Dittrich <dittrich@qualitus.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Chart\Speedo;
+
+/**
+ * Interface Factory
+ * @package ILIAS\UI\Component\Chart\Speedo
+ */
+interface Factory {
+
+    /**
+     * ---
+     * description:
+     *   purpose: The standard speedo is used if the speedo is to be rendered in it's original size.
+     *
+     * rules:
+     *   composition:
+     *     1: Speedos MUST contain a goal value. It MUST be numeric and MAY be a score equal to 100%.

```

Comment:
Not sure about the term "goal", would not maximum be clearer. The second part is not clear to me, I think it means to say:
"It MUST be numeric represents the maximum score."
It MUST be numeric is fine as rule. move the other part (represents the maximum score) to the desription.